sys-apps/portage/portage-9999.ebuild

```diff
@@ -160,17 +171,23 @@ my_src_configure() {
 }
 
 src_compile() {
-	python_foreach_impl meson_src_compile
+	distutils-r1_src_compile
+	BUILD_DIR="${S}-meson" python_foreach_impl meson_src_compile
 }
 
 src_test() {
 	local -x PYTEST_ADDOPTS="-vv -ra -l -o console_output_style=count -n $(makeopts_jobs) --dist=worksteal"
 
-	python_foreach_impl meson_src_test --no-rebuild --verbose
+	BUILD_DIR="${S}-meson" python_foreach_impl meson_src_test --no-rebuild --verbose
 }
 
 src_install() {
-	python_foreach_impl my_src_install
+	distutils-r1_src_install
+	# TODO: Add a setup arg to exclude this stuff.
+	rm -rf "${ED}/usr/"{bin,sbin,etc,lib/portage/bin} || die
+	# Omit excluded files that begin with ../ from RECORD files.
+	find "${ED}" -name RECORD | xargs sed -e '/^\.\.\//d' -i || die
+	BUILD_DIR="${S}-meson" python_foreach_impl my_src_install
```

Comment:
It feels like this is trying very hard to hack how distutils works, simply to acquire the *.dist-info/ directory itself.

What is RECORD needed for here, anyways? Other than nothing. It's essentially pip trying to duplicate a vdb CONTENTS except not very well. It's even optional, per the spec -- in fact, all files in there are optional, other than METADATA.

The RECORD file is actually *bad* to have. It exists solely for pip to know how to uninstall the package using `pip uninstall` which obviously must never happen.

Reply:
Yeah that's true. I've changed it to delete both the RECORD and entry_points.txt files (entry_points.txt  refers to some entry points that do not all exist in the system-wide context).

Reply:
> This seems like extending a hacky reinvention of `distutils-r1` logic in pure `meson` with hacky combination of `distutils-r1` logic with previous hacky reinvention. Can't you just use `DISTUTILS_ARGS` (i.e. effectively meson args), to make meson do what you need it to do in PEP517 mode?

My understanding is that we can't install anything outside of sys.prefix in PEP517 mode (/usr on gentoo), so that would not account for installation of files into /etc. I suppose we could just use basic insinto and doins for /etc files though. How does that sound?

Reply:
We could just use basic insinto / doins for a METADATA file.

Actually possibly you could use install_data inside portage's own meson.build for this...

Reply:
I prefer Eli's idea.

Reply:
I also like Eli's idea to generate a METADATA file and install it with install_data, and we can use the path from [get_install_dir](https://mesonbuild.com/Python-module.html#get_install_dir) to generate the [install_data](https://mesonbuild.com/Reference-manual_functions.html#install_data) install_dir.

Reply:
[meson.build: Install dist-info METADATA](https://github.com/gentoo/portage/pull/1303)